### ISSUE TITLE
DC-funded affordable housing units on opendata.dc.gov

### ISSUE THREAD

**Issue by NealHumphrey** [codefordc/housing-insights#194]
Add the DC-funded buildings to our list of all affordable buildings and subsidies using the [Affordable Housing dataset](http://opendata.dc.gov/datasets/34ae3d3c9752434a8c03aca5deb550eb_62) on opendata.dc.gov. See notes on what is in this data set in #137 . 

**Current related data:**
Our current database contains a 'projects' table, which is pulled from the [Preservation Catalog](www.neighborhoodinfodc.org/dcpreservationcatalog/). In addition, it has a 'subsidy' table, where project -> subsidy is one->many.  This catalog aggregates all the subsidized buildings from the lists published by the federal government; therefore it only contains buildings federally subsidized buildings. It does not systematically include buildings that have only received subsidies from the DC Government; 

**Data adding**
We want to create a row in the project table and in the subsidy table corresponding to each additional building in the opendata.dc.gov dataset. 

There are several ways that a duplicate record could be created, which we want to avoid:

- LIHTC funding is distributed by DHCD; the PresCat includes data from the federal list of LIHTC recipients, so we would expect most of these buildings to be duplicated (and would like to flag any that are not in the federal list but are in the DHCD list. 
- A federally subsidized building can get an additional subsidy/rehab loan from DHCD. In this case, we want to add a record only to the 'subsidy' table and link it to the proper 'project' row using NLHIC_ID
- Occasionally buildings are manually added to the PresCat, so these may already be in the catalog. We want to update any fields in the 'project' table if better data is available from DHCD. 
- There a large number of subsidy records (and corresponding project records) that have the subsidy source "DC/AFFPIPELINE". These are at least 8 years old (though the data for the subsidy is likely still valid) - there may be a few overlapping buildings but for the most part new buildings are not being added to the Preservation Catalog currently. 

Methodology
- [x] Create a DCHousingApiConn class based on the MarApiConn class currently in the code. Use this to add raw data files to /data/raw/tax_assessment/opendata/YYYYMMDD when the python/cmd/data.py script is run with the appropriate arguments. Write a demo command to run at the command line to create this file with the appropriate datestamp.
(pull request after step 1)
- [x] Adjust the relevant meta.json and manifest.csv to add both this new flat file and the most recent preservation_catalog file to the projects table in the database. You'll likely need to use the newly-created 'optional' fields. Related guidance is here:  [adding a new dataset](http://housinginsights.org/resources/adding-data-sources.html). In the DCHousingCleaner, you'll need to supply duplicate entity matching to not duplicate buildings we already know about. 
- [x] For every building in the source data set (whether or not a duplicate entry was found), create a new entry in the subsidy table. We'll need to figure out best way to do this - likely by creating a second flat file during the cleaning stage. The easiest way to do this without big alterations to our current ingestion process is assume that the newly created subsidy file will be available to the ingestion code by placing the subsidy file in the list after the affordable housing file.

**Comment by NealHumphrey** [comment]
@jkwening you should also know about this work that has already been done on some de-duplication scripts. It has not been merged in so it is not in the current code base - I will ask Louis to do a pull request. But, you won't want to use it directly - instead you can pull from it as needed to incorporate into the object-oriented approach we're using now.

**Comment by NealHumphrey** [comment]
PR 233 is merged in! Next up: combining the dchousing table with the projects table. @jkwening let's discuss tonight.

**Comment by NealHumphrey** [comment]
Hey @jkwening Flagging a future issue we'll run into using the endpoint you set up for this. ESRI by default limits the API calls to 1,000 records, as we found in #198 . Using the geojson endpoint but swapping .geojson for .csv will return the full api data already formatted as .csv. We should switch to this data download method - though you can maybe wait to convert the code until we do another api endpoint, as we might end up making an opendata conn base class that does stuff like this.

**Comment by ptatian** [comment]
@jkwening In case it's helpful, here's a basic diagram of the Preservation Catalog file schema. 
https://urbanorg.box.com/s/t1t8cg9228agxdvvqpg61uxdo7qj8gwe

**Comment by jkwening** [comment]
Cleaning up related data ingestion issues - closing because it looks like this has been completed. Note - there are some outstanding issues related to cleaning up encoded data for the project table fields ( #373)

**Comment by NealHumphrey** [comment]
Note, I moved the flagged issue in my May 12 comment into its own issue #374
